File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        int idx = 0;
        
        for(int[] n : queries){
            answer[idx] = 9_999_999;
            for(int i = n[0] ; i <= n[1] ; i++){
                if(n[2] < arr[i]) answer[idx] = Math.min(answer[idx], arr[i]);
            }
            if(answer[idx] == 9_999_999) answer[idx] = -1;
            idx++;
        }
        return answer;
    }
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    for s,e,k in queries:
        num = 10**9
        for i in range(s,e+1):
            if arr[i] > k:
                num = min(arr[i], num)
                
        answer.append(-1 if num == 10**9 else num)
    return answer